### PR TITLE
Improve Reader View performance

### PR DESCRIPTION
**Pull request by rlrh**
Still WIP, but open for reviews.

#### Summary of changes:
- Refactor to simplify code
- Add `getArticle` method to Model, Logic
- In reader view mode, shortcut loading by processing downloaded article content to generate reader view
- If changing styles and reader view is already loaded, just change stylesheet without reloading.

### CODE REVIEW

File: src/main/java/seedu/address/model/ModelManager.java [hunk 1]
```diff
@@ -338,6 +339,19 @@ public void deleteArticle(URL url) throws IOException {
...
 
+    @Override
+    public Optional<String> getArticle(URL url) {
+        return storage
+                .getOfflineLink(url)
+                .map(path -> {
+                    try {
+                        return FileUtil.readFromFile(path);
+                    } catch (IOException ioe) {
+                        return null;
```
Comment (qjqqyy) [inline src/main/java/seedu/address/model/ModelManager.java:350]: did you mean `Optional.empty`?

File: src/main/java/seedu/address/ui/BrowserPanel.java [hunk 2]
```diff
@@ -139,63 +142,97 @@ private void handleSucceeded() {
...
+        loadReaderViewFailurePage();
+    }
+
+    /**
+     * Updates online link whenever a hyperlink is clicked.
+     */
+    private void processHyperlinks() {
+        Optional.ofNullable(webEngine.getDocument())
+                .map(document -> document.getElementsByTagName("a"))
+                .ifPresent(links -> {
```
Comment (qjqqyy) [inline src/main/java/seedu/address/ui/BrowserPanel.java:179]: I'm tempted to golf this....
```java
Optional.ofNullable(webEngine.getDocument())
        .map(document -> document.getElementsByTagName("a")).stream()
        .flatMap(nodeList -> IntStream.range(0, nodeList.getLength()).mapToObj(nodeList::item))
        .forEach(link -> {...});
```

Comment (qjqqyy) [reply src/main/java/seedu/address/ui/BrowserPanel.java:179]: but jokes aside maybe we can name some of the lambdas here to demonstrate more slap or something?

File: src/main/java/seedu/address/model/ModelManager.java (hunk 1 above)
Comment (rlrh) [reply src/main/java/seedu/address/model/ModelManager.java:350]: Nope, returning null in a mapping function will result in `Optional.empty()`.

**Comment by rlrh** [comment]
Outdated - I decided to completely rewrite it.

**Pull request closed without merging**